source/Halibut/Transport/Streams/WriteIntoMemoryBufferStream.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Halibut.Transport.Observability;
+
+namespace Halibut.Transport.Streams
+{
+    public class WriteIntoMemoryBufferStream : Stream
+    {
+        readonly MemoryStream memoryStream;
+        readonly Stream sinkStream;
```

Comment:
Rename to `inner` because all I read is "stinkStream"

Reply:
Lol, sure. I was going for "source and sink", but am happy to rename 😁